### PR TITLE
chore: add release branches to dependabot gomod updates

### PR DESCRIPTION
**Pull request by rbtr**
**Reason for Change**:

Adds release branches to dependabot config for gomod updates.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by jpayne3506 — APPROVED** [review]
As it stands this is good, but because of us pinning release trains to other deps I would recommend looking at [disabling updates for certain versions + deps](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates#example-disabling-version-updates-for-some-dependencies)

**Review by thatmattlong — APPROVED** [review]

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -62,3 +62,43 @@ updates:
   commit-message:
     prefix: "deps"
   open-pull-requests-limit: 10
+- package-ecosystem: "gomod"
+  directory: "/"
```
Comment (huntergregory) [inline .github/dependabot.yaml:66]: curious the logic for having the same line for v1.5 for both `/` and `/azure-ipam`? Assuming the former will cover all directories?

**Comment by rbtr** [comment]
> As it stands this is good, but because of us pinning release trains to other deps I would recommend looking at [disabling updates for certain versions + deps](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates#example-disabling-version-updates-for-some-dependencies)

once we interact with dependabot on a PR it opens, it tracks these things automatically

File: .github/dependabot.yaml (hunk 1 above)
Comment (rbtr) [reply .github/dependabot.yaml:66]: no, this line tells dependabot which directory to look in for the `go.mod`. We need it to update both `/go.mod` and `/azure-ipam/go.mod`

**Pull request merged**